com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java

```diff
@@ -264,14 +264,21 @@ protected void _toJavaStatement(final XIssueExpression expr, final ITreeAppendab
       b.newLine();
     }
 
-    // checking for null context EObject
-    b.append("if (");
+    boolean issueExpressionEqualsImplicitVariable = true;
     if (markerObject != null) {
-      internalToConvertedExpression(markerObject, b, getLightweightType(eObjectType));
-    } else {
-      b.append(getContextImplicitVariableName(expr));
+      issueExpressionEqualsImplicitVariable = markerObject.toString().equalsIgnoreCase(getContextImplicitVariableName(expr));
+    }
+
+    if (!issueExpressionEqualsImplicitVariable) {
+      // checking for null context EObject
+      b.append("if (");
+      if (markerObject != null) {
+        internalToConvertedExpression(markerObject, b, getLightweightType(eObjectType));
+      } else {
+        b.append(getContextImplicitVariableName(expr));

```

Comment:
this no longer needs to be here?

Reply:
specifically the else, as we don't need to test if contextImplicitVariable is null

Reply:
I take it back, we don't need the whole if at line 275, based on what we already have at line 268

Reply:
Thank you for the comment. It is redundant.

Reply:
Done.